test/Sia.Gateway.Tests/Requests/Events/CreatLinksWorksCorrectly_WhenCalledByEventsMethods.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Sia.Gateway.Controllers;
+using Sia.Shared.Protocol;
+using System.Collections.Generic;
+
+namespace Sia.Gateway.Tests.Requests.Events
+{
+    [TestClass]
+    public class GenerateLinksHeaderTestsForEvents
+    {
+        private List<string> methods;
+        private List<object> ids;
+        private Mock<IUrlHelper> urlHelperMock;
+        public GenerateLinksHeaderTestsForEvents()
+        {
+            // Arrange
+            methods = new List<string>();
+            ids = new List<object>();
+            urlHelperMock = new Mock<IUrlHelper>();
+            urlHelperMock.Setup(link => link.Link(It.IsAny<string>(), It.IsAny<object>()))
+                .Callback<string, object>(
+                    (s, o) =>
+                    {
+                        methods.Add(s);
+                        ids.Add(o);
+                    }
+                );
+
+        }
+        
+        string GetProperty(object values, string property) => values.GetType().GetProperty(property)?.GetValue(values).ToString() ?? "";
+
+        [TestMethod]
+        public void CreateLinksWorksCorrectly_WhenCalledInEventsMethods()
```

Comment:
It's often better to be more descriptive in test names; we can assume that any test is verifying that something works correctly, and the test name should include a description of what behavior we consider 'working correctly' to be. Similarly, the 'when' statement should give information about the content of the context of the method call rather than where the method is being called (unless the 'when' materially changes the behavior of the method). This test might be better as 'CreateLinksGeneratesFourLinksWithCorrectIds_WhenPassedAnIncidentIdAndAnEventId()', but there are likely a number of other options at least as descriptive as that. Naming is a very hard problem, but effort put into it pays off over time.

Reply:
CreateLinksGeneratesFourLinksWithCorrectIds_WhenPassedAnIncidentIdAndAnEventId() does make more sense.